tests/test_non_host.py

```diff
@@ -40,10 +40,29 @@ def test_custom_interpreter(
         expected -= {"setuptools", "wheel"}
     assert found == expected, out
 
-    monkeypatch.setattr(sys, "argv", [*cmd, "--graph-output", "something"])
-    with pytest.raises(SystemExit) as context:
-        main()
-    out, err = capfd.readouterr()
-    assert context.value.code == 1
-    assert not out
-    assert err == "graphviz functionality is not supported when querying non-host python\n"
+
+def test_custom_interpreter_with_local_only(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    capfd: pytest.CaptureFixture[str],
+) -> None:
+    venv_path = str(tmp_path / "venv")
+
+    result = virtualenv.cli_run([venv_path, "--system-site-packages", "--activators", ""])
+
+    cmd = ["", f"--python={result.creator.exe}", "--local-only"]
+    monkeypatch.setattr(sys, "prefix", venv_path)
+    monkeypatch.setattr(sys, "argv", cmd)
```

Comment:
Turns out monkey patching is dangerous to do as it actually changes `sys.prefix` and `sys.argv` directly. Let's use `pytest_mock.MockerFixture` instead and mock where they are "looked up" instead (see [here](https://docs.python.org/3/library/unittest.mock.html#id6) for more details). Like this:
```suggestion
   mocker: MockerFixture,
    capfd: pytest.CaptureFixture[str],
) -> None:
    venv_path = str(tmp_path / "venv")

    result = virtualenv.cli_run([venv_path, "--system-site-packages", "--activators", ""])

    cmd = ["", f"--python={result.creator.exe}", "--local-only"]
    mocker.patch("pipdeptree._discovery.sys.prefix", venv_path)
    mocker.patch("pipdeptree._discovery.sys.argv", cmd)
```

Reply:
Done all.

Reply:
I appreciate you looking for other areas where monkeypatch was being misused, but next time please just change the requested areas so that the PR isn't trying to do too many things at once; another PR could be opened for the monkeypatch misuse 👍